entity-management-web/src/main/java/eu/europeana/entitymanagement/web/service/EntityRecordService.java

```diff
@@ -293,22 +293,22 @@ private String generateEntityId(String entityType, String entityId) {
     }
 
     /**
-     * Checks if any of the resources in the SameAs field from the Datasource is already
+     * Checks if any of the resources in the SameAs / ExactMatch field from the Datasource is already
      * known.
      * 
-     * @param rdfResources list of SameAs resources
+     * @param rdfResources list of SameAs/ExactMatch resources
      * @return Optional containing EntityRecord, or empty Optional if none found
      */
-    public Optional<EntityRecord> retrieveCoreferenceSameAs(List<String> rdfResources) {
-	for (String resource : rdfResources) {
-	    Optional<EntityRecord> entityRecordOptional = retrieveByEntityId(resource);
-	    if (entityRecordOptional.isPresent()) {
-		return entityRecordOptional;
-	    }
-	}
+		public Optional<EntityRecord> retrieveCoreferencedEntity(List<String> rdfResources) {
+			for (String resource : rdfResources) {
+				Optional<EntityRecord> entityRecordOptional = retrieveByEntityId(resource);
```

Comment:
this method should send the database requests to search in sameAs/exactMatch fields. It currently tries to read by entityId. Additionally this method should send only one database request using $in operator